### PR TITLE
[macOS] Add `platform_view` example.

### PR DESCRIPTION
**Pull request by a-wallen**
Adds an implementation for `platform_view` on macOS for better example coverage and for device lab tests.

https://user-images.githubusercontent.com/44445638/188499829-99810321-d7ca-436b-a743-920f50c68bab.mov

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by yaakovschectman — APPROVED** [review]
LGTM

**Review by loic-sharma — APPROVED** [review]
Looks reasonable to me but consider getting approval from someone that's more familiar with macOS!

**Review by jmagman — APPROVED** [review]

File: examples/platform_view/macos/Runner/PlatformViewController.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Cocoa
+
+class PlatformViewController: NSViewController {
```
Comment (cbracken) [inline examples/platform_view/macos/Runner/PlatformViewController.swift:7]: nit: add a doc comment, especially since it's an example.

**Review by cbracken — COMMENTED** [review]
lgtm with one nit!

**Pull request merged**